src/plugins/broadcaster.ts

```diff
@@ -0,0 +1,47 @@
+import { Plugin, Cordova } from './plugin';
+
+/**
+ * @name Broadcaster
+ * @description
+ * This plugin adds exchanging events between native code and your app.
+ *
+ * @usage
+ * ```
+ * import { Broadcaster } from 'ionic-native';
+ *
+ * // Listen to events from Native
+ * Broadcaster.addEventListener('eventName').then((event) => console.log(event));
+ *
+ * // Send event to Native
+ * Broadcaster.fireNativeEvent('eventName', {}).then(() => console.log('success'));
+ *
+ * ```
+ */
+@Plugin({
+  pluginName: 'Broadcaster',
+  plugin: 'cordova-plugin-broadcaster',
+  pluginRef: 'broadcaster',
+  repo: 'https://github.com/bsorrentino/cordova-broadcaster',
+  platforms: ['Android', 'iOS'],
+  install: 'ionic plugin add cordova-plugin-broadcaster'
+})
+export class Broadcaster {
+
+  /**
+   * This function listen to an event sent from the native code
+   * @param eventName {string}
+   * @return {Promise<any>} Returns a promise that resolves when an event is received
+   */
+  @Cordova()
+  static addEventListener(eventName: string): Promise<any> { return; }
```

Comment:
A promise resolves once, an observable can fire as many times as you want until you unsubscribe from it.

It should also have the following decorator instead of just `observable: true`, this will make sure we unsubscribe from the event when unsubscribing from the observable:
```
@Cordova({
  observable: true,
  clearFunction: 'removeEventListener',
  clearWithArgs: true
})
```